devices/flash-zynq7000/flashdrv.c

```diff
@@ -420,6 +447,72 @@ static ssize_t flashdrv_write(unsigned int minor, addr_t offs, const void *buff,
 }
 
 
+static ssize_t flashdrv_erase(unsigned int minor, addr_t addr, size_t len, unsigned int flags)
+{
+	int res;
+	size_t sectSz;
+	unsigned int id = 0;
+	addr_t end, addr_mask;
+	const flash_cfi_t *cfi = &fdrv_common.info.cfi;
+
+	if (len == 0) {
+		return 0;
+	}
+
+	res = flashdrv_regionFind(addr, &id);
+	if (res < 0) {
+		return res;
+	}
+
+	sectSz = CFI_SIZE_SECTION(cfi->regs[id].size);
+
+	addr_mask = sectSz - 1;
+	end = (addr + len + addr_mask) & ~addr_mask;
```

Comment:
There's no `len` sanitization - it case of `len` greater then size of the memory I guess it'll work, sectors would be erased, but we'll exit with an error

Reply:
Done